app/scripts/inpage.js

```diff
@@ -65,26 +65,28 @@ initProvider({
 
 // setup web3
 
-if (typeof window.web3 !== 'undefined') {
-  throw new Error(`MetaMask detected another web3.
-     MetaMask will not work reliably with another web3 extension.
+if (typeof window.ethereum !== 'undefined' || typeof window.web3 !== 'undefined') {
+  log.warn(`MetaMask detected another ethereum provider.
+     MetaMask will not work reliably with another wallet present.
      This usually happens if you have two MetaMasks installed,
      or MetaMask and another web3 extension. Please remove one
      and try again.`)
-}
+} else {
 
-const web3 = new Web3(window.ethereum)
-web3.setProvider = function () {
-  log.debug('MetaMask - overrode web3.setProvider')
-}
-log.debug('MetaMask - injected web3')
+  const web3 = new Web3(window.ethereum)
+  web3.setProvider = function () {
+    log.debug('MetaMask - overrode web3.setProvider')
+  }
+  log.debug('MetaMask - injected provider')
 
-Object.defineProperty(window.ethereum, '_web3Ref', {
-  enumerable: false,
-  writable: true,
-  configurable: true,
-  value: web3.eth,
-})
+  Object.defineProperty(window.ethereum, '_web3Ref', {
+    enumerable: false,
+    writable: true,
+    configurable: true,
+    value: web3.eth,
+  })
+
+  // setup dapp auto reload AND proxy web3
+  setupDappAutoReload(web3, window.ethereum._publicConfigStore)
+}
```

Comment:
When this is rebased, the `if` condition should just check if `window.web3` is undefined, and the `if` clause should  `log.warn` that another `window.web3` was detected.

The `else` clause should call `setupWeb3`, per the current state of the file: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/inpage.js

Reply:
Implemented this change but also detect for existing ethereum provider before performing that setup.